tests/unit/sys/test_expressionparser.cxx

```diff
@@ -213,20 +213,25 @@ TEST_F(ExpressionParserTest, BadNumbers) {
 }
 
 TEST_F(ExpressionParserTest, BadFunctions) {
-  EXPECT_THROW(parser.parseString("sin(--)"), ParseException);
-  EXPECT_THROW(parser.parseString("sin(x, )"), ParseException);
-  EXPECT_THROW(parser.parseString("sin(x = 1)"), ParseException);
+  parser.addGenerator("increment", std::make_shared<IncrementGenerator>());
+
+  EXPECT_THROW(parser.parseString("increment(--)"), ParseException);
+  EXPECT_THROW(parser.parseString("increment(x, )"), ParseException);
+  EXPECT_THROW(parser.parseString("increment(x = 1)"), ParseException);
 }
 
 TEST_F(ExpressionParserTest, BadExpressions) {
+  parser.addGenerator("increment", std::make_shared<IncrementGenerator>());
+
   EXPECT_THROW(parser.parseString("x = 1"), ParseException);
-  EXPECT_THROW(parser.parseString("sin(x"), ParseException);
-  EXPECT_THROW(parser.parseString("sin[x"), ParseException);
+  EXPECT_THROW(parser.parseString("increment(x"), ParseException);
+  EXPECT_THROW(parser.parseString("increment"), ParseException);
```

Comment:
This has also lost the `[` as well as changing `sin`-->`increment`, is that intentional?

Reply:
Yes, the `[` was extraneous for hitting that exception (i.e. it was already throwing before parsing the `[`)